index.js

```diff
@@ -71,12 +71,6 @@ function compressPlugin (fastify, opts, next) {
       return this.send(payload)
     }
 
-    if (encoding === null) {
```

Comment:
I would leave this feature as opt-in since it could be willing for the user to choose what to do since the RFC say "should" and not "must".

We should choose also the default in order to decide if this fix will be semver-major or not

Reply:
I’ve read the spec, and I do not see a good reason for the old behavior to be there.

Reply:
The reason would be to apply the 406 logic also to the encoding and be apply strict validation to the requests like the other headers. The [`should` definition](https://tools.ietf.org/html/rfc2119) may consent this behaviour that we are removing. _To be even more open this logic could be an optional callback with the 406 logic_

Of course, this must be interpreted as an additional feature not a standard in any way.

Reply:
I agree that it would be nice to allow the user to choose how they want to deal with the case where the encoding is invalid, but in no way is it required, and I can't imagine too many people would want to override standard behavior that we *should* have (defined by IETF).

I could see having an optional callback in the plugin options and the compress function arguments. @mcollina what do you think of this?